packages/backend/src/nest/storage/communityMetadata/communityMetadata.store.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { jest, beforeEach, describe, it, expect, afterEach, beforeAll, test } from '@jest/globals'
+import OrbitDB from 'orbit-db'
+import { EventEmitter } from 'events'
+
+import { CommunityMetadataStore } from './communityMetadata.store'
+import { LocalDbService } from '../../local-db/local-db.service'
+
+const mockOrbitDb = {
+  identity: { id: 'orbitDbId', provider: jest.fn() },
+  keyvalue: jest.fn(
+    (): { load: jest.Mock, get: jest.Mock, put: jest.Mock, events: { on: jest.Mock }, all: object } => {
+      return {
+        load: jest.fn(),
+        get: jest.fn(),
+        put: jest.fn(),
+        events: { on: jest.fn() },
+        all: {}
+      }
+    }
+  ),
+}
+const mockLocalDbService = { putOwnerOrbitDbIdentity: jest.fn(), getOwnerOrbitDbIdentity: jest.fn() }
+const mockEmitter = { emit: jest.fn() }
+
+describe('CommmunityMetadataStore/updateCommunityMetadata', () => {
+  test('does not update community metadata if the metadata is invalid', async () => {
+    const meta = {
+      id: 'anId',
+      rootCa: 'CA time!',
+      ownerCertificate: 'Certificately me',
+    }
+
+    const store = new CommunityMetadataStore()
+    await store.init(mockOrbitDb as unknown as OrbitDB, mockLocalDbService as unknown as LocalDbService, mockEmitter as unknown as EventEmitter)
```

Comment:
I'm not entirely sure what you mean. Can you expand on this idea a little bit? Thanks!

Reply:
My idea is the solution I posted simplifies the types and is consistent between mine and Barteks' tests

Reply:
Ok, would you like to see this done in this PR or is it sort of like we could do it but not the most important thing and could potentially be done later? I'm curious to hear what you think is important to change before merging.

Reply:
My experience tells me if we didn't do something right away we never do it

Reply:
Ok, you'd like to do it now? I think it can be difficult to tell what is a suggestion, what is a question for collaboration and what is a requirement for approval. I don't like to assume too much when conversing, especially over text, but it sounds like you'd like to see this changed in this PR, so I can look into it changing it. Thanks for sharing and discussing with me, I appreciate it